### DISCUSSION TITLE
Why exactly do I need Git?

### DISCUSSION THREAD

**Discussion by DNin01** [ScratchAddons/ScratchAddons#4940] [Help] [answered]
The [contributing guidelines](https://github.com/ScratchAddons/ScratchAddons/blob/97fe3d741c9d200bc03cf8098d134aaafc12b1a8/CONTRIBUTING.md) say that contributors should be familiar with Git.
Why is that? (Is there something I have to do with it? Initiate checks? Something else?)
Who would be using it and when? (Is this only necessary for [the listed contributors](https://github.com/ScratchAddons/ScratchAddons/graphs/contributors)/becoming a contributor or is this for anyone making pull requests?)
If this is advice for anyone making pull requests, do we _have_ to know Git? (Is it only a suggestion?)

I don't have any experience with Git and I also don't have it installed. Can I still make a pull request, or should I probably learn Git first?

**Edit:** Just figured out that you don't have to do any transfers when you make a new branch, GitHub can just copy everything over from the source branch that's already selected (and that's probably also what happens if I were to use `git checkout -b new-branch`). Now the question is, how would I synchronize stuff (you know, make sure my fork is up to date) with just the web UI (I'd much rather use the web UI or, if necessary, GitHub Desktop)? Any tips on that? I don't have any experience with managing forks, and I don't really want to get started until I can work with a fork effectively and make sure everything goes smoothly.

If I have to use Git, I'll just have to do some learning. It sounds like a lot to me now, but I know it's not going to stay hard, I'd just need to find a good guide and write down the instructions, plus, I shouldn't need to know every single command.

**Comment by WorldLanguages** [comment]
How would you send a pull request without Git?

**Reply by Norbiros** [reply]
You don't need to learn git, but it is better to know that... You can use gitlab, but you need to understand branches, commits, PRS etc. 

If you don't want to install anything, use github vscode online, but it is better to use it with your IDE (or with CLI)

**Reply by rimopa** [reply]
I only use GitHub Desktop and it works for me.

**Reply by WorldLanguages** [reply]
> I only use GitHub Desktop and it works for me.

GitHub Desktop uses git under the hood
Of course GUIs that use git internally can be used just fine

**Reply by Norbiros** [reply]
Git can you give more possibilities, but for normal things gitlab is great alternative (better than github code editor).

**Reply by DNin01** [reply]
> How would you send a pull request without Git?

Can't I do it from [here](https://github.com/ScratchAddons/ScratchAddons/compare/master...WorldLanguages:ScratchAddons:master), for example?

**Reply by WorldLanguages** [reply]
My message was confusing, sorry - most of us do create the pull requests through the GitHub web GUI.
I meant how you would commit and push the changes to your own branch without Git.

**Reply by DNin01** [reply]
> My message was confusing, sorry - most of us do create the pull requests through the GitHub web GUI. I meant how you would commit and push the changes to your own branch without Git.

So, is Git what you would use to copy the content from the master branch into your own or something like that?

(I just haven't ever done this before)

**Reply by Norbiros** [reply]
> > My message was confusing, sorry - most of us do create the pull requests through the GitHub web GUI. I meant how you would commit and push the changes to your own branch without Git.
> 
> So, is Git what you would use to copy the content from the master branch into your own or something like that?
> 
> (I just haven't ever done this before)

Something like that... You pull changes from this origin repo (`git pull`), then you create new branch (`git checkout -b fix-sa-bug`), then you made your changes to the files, and if you want to commit you add files to commit (`git add .`) and then you create commit (`git commit -m "I fixed very important bug" `), after that you push your changes (`git push`)

**Reply by DNin01** [reply]
Just figured out that you don't have to do any transfers when you make a new branch, GitHub can just copy everything over from the source branch that's already selected (and that's probably also what happens if I were to use `git checkout -b new-branch`). Now the question is, how would I synchronize stuff (you know, make sure my fork is up to date) with just the web UI (I'd much rather use the web UI or, if necessary, GitHub Desktop)? Any tips on that? I don't have any experience with managing forks, and I don't really want to get started until I can work with a fork effectively and make sure everything goes smoothly.

If I have to use Git, I'll just have to do some learning. It sounds like a lot to me now, but I know it's not going to stay hard, I'd just need to find a good guide and write down the instructions, plus, I shouldn't need to know every single command.

**Reply by Samq64** [reply]
I recommend just downloading GH Dsktop.

**Comment by DNin01** [comment, marked as answer]
Oh, GitHub Docs shows that [syncing a fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork) can be done with the web UI! And I know that I can create branches and pull requests with it too, so I think I'm all set. I'll comment again if I have any more questions.

Hopefully, I won't mind downloading GitHub Desktop if I need it for anything. It's pretty cool.

<sub>Here's hoping I don't miss anything else on my first pull request, like making sure the checks work...</sub>
